test/tuple.jl

```diff
@@ -1,7 +1,50 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-@test convert(Tuple, (1,2)) == (1,2)
-@testset "indexing" begin
+struct BitPerm_19352
+    p::NTuple{8,UInt8}
+    function BitPerm(p::NTuple{8,UInt8})
+        sort(collect(p)) != collect(0:7) && error("$p is not a permutation of 0:7")
+        new(p)
+    end
+    BitPerm_19352(xs::Vararg{Any,8}) = BitPerm(map(UInt8, xs))
```

Comment:
this is a bit of a refactoring of the constructor, hopefully it would still reproduce the original issue in this form?